server/src/main/java/org/jboss/as/server/ServerService.java

```diff
@@ -433,6 +433,9 @@ public void undeploy(DeploymentUnit context) {
             }
             String message = ServerLogger.ROOT_LOGGER.unsuccessfulBoot(String.join(" ", messages));
             bootstrapListener.bootFailure(new Exception(message, cause));
+            if (cause != null) {
+                cause.printStackTrace();
+            }
```

Comment:
I'm pretty sure this fixes the test failure, but I need to understand why still. I'll have to do some testing against the main branch with logging 3.5.3 to see why this doesn't cause an issue there.